### ISSUE TITLE
dial tcp 127.0.0.1:4646: connect: connection refused

### ISSUE THREAD

**Issue by drain** [hashicorp/nomad#11755]
### Nomad version
Nomad v1.2.3 (a79efc8422082c4790046c3f5ad92c542592a54f)

### Operating system and Environment details
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal

### Issue
Error joining: failed joining: Put "http://127.0.0.1:4646/v1/agent/join?address=64.225.18.170%3A4648": dial tcp 127.0.0.1:4646: connect: connection refused

### Reproduction steps
Start a dev agent and attempt to cluster a node

#### Expected Result
Node connects to cluster

#### Actual Result
Error

### Nomad Server logs (if appropriate)
```
==> No configuration files loaded
==> Starting Nomad agent...
==> Nomad agent configuration:

       Advertise Addrs: HTTP: 10.17.0.30:4646; RPC: 10.17.0.30:4647; Serf: 10.17.0.30:4648
            Bind Addrs: HTTP: 0.0.0.0:4646; RPC: 0.0.0.0:4647; Serf: 0.0.0.0:4648
                Client: true
             Log Level: INFO
                Region: global (DC: dc1)
                Server: true
               Version: 1.2.3

==> Nomad agent started! Log data will stream in below:

    2021-12-28T18:47:38.713Z [INFO]  agent: detected plugin: name=raw_exec type=driver plugin_version=0.1.0
    2021-12-28T18:47:38.713Z [INFO]  agent: detected plugin: name=exec type=driver plugin_version=0.1.0
    2021-12-28T18:47:38.713Z [INFO]  agent: detected plugin: name=qemu type=driver plugin_version=0.1.0
    2021-12-28T18:47:38.713Z [INFO]  agent: detected plugin: name=java type=driver plugin_version=0.1.0
    2021-12-28T18:47:38.713Z [INFO]  agent: detected plugin: name=docker type=driver plugin_version=0.1.0
    2021-12-28T18:47:38.716Z [INFO]  nomad.raft: initial configuration: index=1 servers="[{Suffrage:Voter ID:10.17.0.30:4647 Address:10.17.0.30:4647}]"
    2021-12-28T18:47:38.716Z [INFO]  nomad: serf: EventMemberJoin: digitalocean-ubuntu-nyc3-s-1vcpu-1gb.global 10.17.0.30
    2021-12-28T18:47:38.716Z [INFO]  nomad: starting scheduling worker(s): num_workers=1 schedulers=["service", "batch", "system", "sysbatch", "_core"]
    2021-12-28T18:47:38.717Z [INFO]  client: using state directory: state_dir=/tmp/NomadClient1315105437
    2021-12-28T18:47:38.717Z [INFO]  client: using alloc directory: alloc_dir=/tmp/NomadClient2050330494
    2021-12-28T18:47:38.717Z [INFO]  client: using dynamic ports: min=20000 max=32000 reserved="\"\""
    2021-12-28T18:47:38.718Z [INFO]  client.cpuset_manager: initialized cpuset cgroup manager: parent=/nomad cpuset=0
    2021-12-28T18:47:38.720Z [INFO]  client.fingerprint_mgr.cgroup: cgroups are available
    2021-12-28T18:47:38.723Z [INFO]  nomad.raft: entering follower state: follower="Node at 10.17.0.30:4647 [Follower]" leader=
    2021-12-28T18:47:38.723Z [INFO]  nomad: adding server: server="digitalocean-ubuntu-nyc3-s-1vcpu-1gb.global (Addr: 10.17.0.30:4647) (DC: dc1)"
    2021-12-28T18:47:38.728Z [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=lo
    2021-12-28T18:47:38.729Z [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=lo
    2021-12-28T18:47:38.730Z [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=eth0
    2021-12-28T18:47:38.732Z [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/ethtool device=eth1
    2021-12-28T18:47:38.767Z [INFO]  client.plugin: starting plugin manager: plugin-type=csi
    2021-12-28T18:47:38.768Z [INFO]  client.plugin: starting plugin manager: plugin-type=driver
    2021-12-28T18:47:38.768Z [INFO]  client.plugin: starting plugin manager: plugin-type=device
    2021-12-28T18:47:38.772Z [INFO]  client: started client: node_id=5be9818e-e7dc-ad69-eaeb-e406f88dc2c9
    2021-12-28T18:47:39.877Z [WARN]  nomad.raft: heartbeat timeout reached, starting election: last-leader=
    2021-12-28T18:47:39.877Z [INFO]  nomad.raft: entering candidate state: node="Node at 10.17.0.30:4647 [Candidate]" term=2
    2021-12-28T18:47:39.878Z [INFO]  nomad.raft: election won: tally=1
    2021-12-28T18:47:39.878Z [INFO]  nomad.raft: entering leader state: leader="Node at 10.17.0.30:4647 [Leader]"
    2021-12-28T18:47:39.878Z [INFO]  nomad: cluster leadership acquired
    2021-12-28T18:47:39.880Z [INFO]  nomad.core: established cluster id: cluster_id=6c476d1c-d473-487f-7251-5e8aa3aefe38 create_time=1640717259880141025
    2021-12-28T18:47:40.035Z [INFO]  client: node registration complete
    2021-12-28T18:47:41.037Z [INFO]  client: node registration complete
```

**Comment by drain** [comment]
tldr read docs
